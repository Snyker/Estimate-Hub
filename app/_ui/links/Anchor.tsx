import Link, {LinkProps} from "next/link";
import React, {PropsWithChildren} from "react";

export const Anchor: React.FC<PropsWithChildren<Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, keyof LinkProps> & LinkProps>> = ({children, className = '', ...props}) => {
    return (
        <Link className={`link link-underline-hover link-primary text-sm ${className}`} {...props}>{children}</Link>
    );
};
