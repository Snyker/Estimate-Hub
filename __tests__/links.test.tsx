import React from 'react';
import renderer from 'react-test-renderer';
import {Anchor} from "@/app/_ui/links/Anchor";
import {LinkProps} from "next/link";

describe('Anchor Component', () => {
    const defaultProps: LinkProps = {
        href: '/example',
    };

    it('renders correctly with default props', () => {
        const tree = renderer.create(<Anchor {...defaultProps}>Click me</Anchor>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('applies custom class', () => {
        const tree = renderer.create(<Anchor {...defaultProps} className="custom-class">Click me</Anchor>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('applies next/link properties correctly', () => {
        const tree = renderer.create(<Anchor {...defaultProps} prefetch={false}>Click me</Anchor>).toJSON();
        expect(tree).toMatchSnapshot();
    });

});
