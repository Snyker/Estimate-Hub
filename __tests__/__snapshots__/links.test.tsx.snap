// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor Component applies custom class 1`] = `
<a
  className="link link-underline-hover link-primary text-sm custom-class"
  href="/example"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  Click me
</a>
`;

exports[`Anchor Component applies next/link properties correctly 1`] = `
<a
  className="link link-underline-hover link-primary text-sm "
  href="/example"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  Click me
</a>
`;

exports[`Anchor Component renders correctly with default props 1`] = `
<a
  className="link link-underline-hover link-primary text-sm "
  href="/example"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  Click me
</a>
`;
